pkg/storaged/utils.js

```diff
@@ -849,10 +925,12 @@ export function teardown_active_usage(client, usage) {
     // physical volumes here, and it is easiest to catch this
     // condition upfront by reshuffling the data structures.
 
-    function unmount(mounteds) {
-        return Promise.all(mounteds.map(m => {
-            return client.unmount_at(m.location, m.users);
-        }));
+    async function unmount(mounteds) {
+        for (const m of mounteds) {
+            await client.unmount_at(m.location, m.users);
+            if (m.set_noauto)
```

Comment:
What is the rationale behind this change? If a user umount's he doesn't want to have a device auto mount anymore?

Reply:
Yes, when you unmount something with Cockpit, this means you don't want this to be mounted anymore, and rebooting shouldn't change that.

When you delete something with a filesystem on it, this will also unmount it, but it will immediately remove the fstab entry as well. So there is no need to add "noauto" to fstab during deletion.

However, when a filesystem needs to be unmounted because its parent is being deleted, then we _should_ add noauto to its fstab entry. (Since, of course, that entry will not be deleted.)